### PR TITLE
[WEB-1644] summary stat percentage rounding

### PR DESCRIPTION
**Pull request by krystophv**
For [WEB-1644], applies a number of rounding rules to prevent rounding display values into filter boundaries.

[WEB-1644]: https://tidepool.atlassian.net/browse/WEB-1644?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/components/clinic/BgRangeSummary.js [hunk 1]
```diff
@@ -17,23 +17,72 @@ import { space, shadows } from '../../themes/baseTheme';
...
+  }
 
-    return format(`.${precision}f`)(percentage);
+  switch (key) {
+    case 'veryLow':
+      if (percentage > 0 && percentage < 0.5) {
+        precision = 2;
+        percentage = roundUp(percentage, precision);
+      }
+      if (percentage > 0.5 && percentage < 1) {
```
Comment (clintonium-119) [inline app/components/clinic/BgRangeSummary.js:46]: I realize what you have matches the done criteria of the ticket, but, would this be better as:
```suggestion
      if (percentage >= 0.5 && percentage < 1) {
```
If I'm tracking this correctly, a `percentage` value of _exactly_ `0.5` would end up with a precision of `0`, and would be formatted as `0` (I don't think `d3-format` rounds for the`f` notation)

If this is correct, we should also update the done criteria of the ticket to suit.

**Review by clintonium-119 — CHANGES_REQUESTED** [review]

File: app/components/clinic/BgRangeSummary.js (hunk 1 above)
Comment (krystophv) [reply app/components/clinic/BgRangeSummary.js:46]: Ah, yeah - good catch. I'll make the adjustment and add a unit test for that case.

Comment (krystophv) [reply app/components/clinic/BgRangeSummary.js]: @clintonium-119 added in the unit test, adjusted the boundary condition and updated the done criteria 👍🏼

**Review by clintonium-119 — APPROVED** [review]
Looks great :fire: 
Approved.
